lib/src/main/java/com/auth0/jwt/JWTVerifier.java

```diff
@@ -371,8 +377,8 @@ private void assertValidClaim(Claim claim, String claimName, Object value) {
             isValid = value.equals(claim.asBoolean());
         } else if (value instanceof Double) {
             isValid = value.equals(claim.asDouble());
-        } else if (value instanceof Date) {
```

Comment:
since we delegate custom `Date` expectations to `withClaim(String name, Instant value)`, we can remove `Date`-based check.